docs/API.md

```diff
@@ -337,6 +337,52 @@ environment variables during the build, you can bring them back via
 * `cargoAuditExtraArgs`
 * `cargoExtraArgs`
 
+### `craneLib.cargoDeny`
+`cargoDeny :: set -> drv`
+
+Create a derivation which will run a `cargo deny` invocation in a cargo
+workspace.
+
+Note that although `cargo deny` can serve as a replacement for `cargo audit`,
+`craneLib.cargoDeny` does not expose this functionality because `cargo deny`
+requires the full source tree, rather than working from just the `Cargo.lock`
+file, meaning it will be re-run when any source file changes, rather than only
+when dependencies change.
+
+Except where noted below, all derivation attributes are delegated to
+`mkCargoDerivation`, and can be used to influence its behavior.
+* `buildPhaseCargoCommand` will be set to run
+  `cargo deny --offline check bans licenses sources`
+  in the workspace.
+* `pnameSuffix` will be set to `"-deny"`
+* `src` will be filtered to only keep `Cargo.lock` files
+
+#### Required attributes
+* `cargoArtifacts`: A path (or derivation) which contains an existing cargo
+  `target` directory, which will be reused at the start of the derivation.
+  Useful for caching incremental cargo builds.
+  - This can be prepared via `buildDepsOnly`
+  - Alternatively, any cargo-based derivation which was built with
+    `doInstallCargoArtifacts = true` will work as well
```

Comment:
Do we actually need artifacts or does `cargo-deny` only need the source? (If they aren't needed, we would also need to change `cargoDeny` to set `cargoArtifacts = null;`)

Reply:
I first tried with all combinations of `cargoArtifacts = null`, `cargoVendorDir = null`, and `doInstallCargoArtifacts = false`; having any of those set in the attrset passed to `mkCargoDerivation` caused the `cargo deny` command to try fetching the crates package index.

Reply:
We only need `cargoArtifacts = null`; `cargoVendorDir` is okay to leave blank (it will automatically be set if not defined) since we actually need it available to check the dependency licenses

Reply:
So the magic combination is `cargoArtifacts = null` and `doInstallCargoArtifacts = false`.  With just `cargoArtifacts = null`, the install phase kicks in right after the `cargo deny` command completes.

Tail end of the Nix log.
```console
++ command cargo --offline deny check bans licenses sources
bans ok, licenses ok, sources ok
@nix { "action": "setPhase", "phase": "installPhase" }
installing
installing target to /nix/store/r3pimy93x88wcnzbli5c02mp9h85qx2n-simple-deny-0.1.0
mv: cannot stat 'target': No such file or directory
```